python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -855,6 +862,50 @@ def with_resources(self, resource_defs: Mapping[str, ResourceDefinition]) -> "As
             freshness_policies_by_key=self.freshness_policies_by_key,
         )
 
+    def copy_for_configured(
+        self,
+        name: str,
+        description: Optional[str],
+        config_schema: IDefinitionConfigSchema,
+        config_or_config_fn: Any,
+    ) -> "AssetsDefinition":
+        from dagster._core.definitions.graph_definition import GraphDefinition
+
+        if isinstance(self.node_def, GraphDefinition):
+            raise DagsterInvalidInvocationError(
+                "Cannot call `.configured` on a graph asset. Instead, call `AssetsDefinition.from_graph(your_graph.configured(...))`"
+            )
+        configured_asset = AssetsDefinition(
+            keys_by_input_name=self._keys_by_input_name,
+            keys_by_output_name=self._keys_by_output_name,
+            node_def=self.node_def.copy_for_configured(
+                name, description, config_schema, config_or_config_fn
+            ),
+            partitions_def=self._partitions_def,
+            partition_mappings=self._partition_mappings,
+            asset_deps=self._asset_deps,
+            selected_asset_keys=self._selected_asset_keys,
+            can_subset=self._can_subset,
+            resource_defs=self._resource_defs,
+            group_names_by_key=self.group_names_by_key,
+            metadata_by_key=self._metadata_by_key,
+            freshness_policies_by_key=self.freshness_policies_by_key,
+        )
+        if len(configured_asset.keys) == 1 and self.node_def.name.endswith(
```

Comment:
does `self.node_def.name.endswith(configured_asset.key.path[-1])` always guarantee equality? It seems like there's some (admittedly pathological) cases where this might let through assets that were created by using `AssetsDefinition` manually rather than using the `@asset` decorator.